backend/taggle-web/src/main/java/kr/taggle/common/config/EtagConfig.java

```diff
@@ -0,0 +1,37 @@
+package kr.taggle.common.config;
+
+import javax.servlet.Filter;
+
+import org.springframework.boot.web.servlet.FilterRegistrationBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.CacheControl;
+import org.springframework.web.filter.ShallowEtagHeaderFilter;
+import org.springframework.web.servlet.config.annotation.ResourceHandlerRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+@Configuration
+public class EtagConfig implements WebMvcConfigurer {
+
+    private static final int MINUTE = 60;
+    private static final int HOUR = 60;
+    private static final int DAY = 24;
+    private static final int YEAR = 365;
+
+    @Override
+    public void addResourceHandlers(ResourceHandlerRegistry registry) {
+        registry.addResourceHandler("/**")
+                .addResourceLocations("classpath:/static/")
+                .setCachePeriod(MINUTE * HOUR * DAY * YEAR)
+                .setCacheControl(CacheControl.noCache().mustRevalidate().cachePublic());
+    }
+
+    @Bean
+    public FilterRegistrationBean filterRegistrationBean(){
+        FilterRegistrationBean registration = new FilterRegistrationBean();
+        Filter etagHeaderFilter = new ShallowEtagHeaderFilter();
+        registration.setFilter(etagHeaderFilter);
+        registration.addUrlPatterns("/*");
+        return registration;
+    }
```

Comment:
```suggestion
    @Bean
    public FilterRegistrationBean<ShallowEtagHeaderFilter> filterRegistrationBean(){
        FilterRegistrationBean<ShallowEtagHeaderFilter> registration = new FilterRegistrationBean<>(new ShallowEtagHeaderFilter());
        registration.addUrlPatterns("/*");
        return registration;
    }
```
의미가 같아 보입니다.(물론 라면 코드는 나중에 커스텀 하기 편할 것 같습니다!!) 이렇게 수정하면 더 간결하고, 인텔리제이 경고문도 사라집니다. 이렇게 수정했을 때 테스트 코드도 정상적으로 동작합니다. 저렇게 작성한 이유가 있는지 궁금합니다. 만약 이유가 없다면 이렇게 수정하는 것이 좋을 것 같아요!!!

Reply:
Etag 미션 요구사항에 있던 예제코드를 참고 해서 작성했습니다.
에러가뜨는건 확인했는데 일단 동작하는것에만 신경 쓰다보니 간과하고 넘어갔네요ㅠㅠ
피드백 감사합니다.